### DISCUSSION TITLE
Any possibility to read DV meta within MKV without demuxing?

### DISCUSSION THREAD

**Discussion by nateify** [quietvoid/dovi_tool#220] [General]
I often download MKV rips online with DV HEVC streams, would it ever be possible to read DV related data directly from the HEVC track inside MKV without spending time to demux or decode in ffmpeg?

**Comment by quietvoid** [comment]
Pipe the stream using FFmpeg like the examples in the README show.
It would only parse the stream, no decoding.

**Reply by nateify** [reply]
To my knowledge this still involves decoding.

I tried the following type of command: `ffmpeg -i file.mkv -c:v copy -vbsf hevc_mp4toannexb -f hevc - | dovi_tool extract-rpu - -o RPU.bin` and on a ~85GB file, this takes more than 10 minutes for my system.

In ffmpeg, `ffmpeg -i input -f null -` is used to test decoding for example.

**Reply by quietvoid** [reply]
No, it only parses and doesn't decode.
In the end the speed is limited by your I/O.

**Reply by NeedsMoar** [reply]
> No, it only parses and doesn't decode. In the end the speed is limited by your I/O.

Yup.  The only thing taking up CPU is the demux which doesn't need much and the parsing likewise.  On a system with enough ram Windows will ram-cache (standby) the most recently ripped file(s) as long as there's space nothing else wants and a pipe command like that has ffmpeg showing FPS in scientific notation and completes as fast as ffmpeg can demux the file.
